### PR TITLE
Bgrm - A terminal application that removes backgrounds from a webcam feed and replaces it with an image

### PR DESCRIPTION
**Pull request by blueOkiris**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [ ] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*:
  + App needs web cam access, so I'm using --device=all
  + App needs to read user images to replace web cam background, so I'm using --filesystem=host:ro
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://github.com/blueOkiris/bgrm
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: bgrm/bgrm.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env python3
```
Comment (barthalion) [inline bgrm/bgrm.py:1]: You should host the source code in your own repository.

**Comment by barthalion** [comment]
bot, build com.blueokiris.bgrm

File: com.blueokiris.bgrm.desktop [hunk 2]
```diff
@@ -0,0 +1,9 @@
+[Desktop Entry]
+Name=bgrm
+Exec=/usr/bin/flatpak run com.blueokiris.bgrm %U
```
Comment (hfiguiere) [inline com.blueokiris.bgrm.desktop:3]: it is NEVER this.
```suggestion
Exec=bgrm-fp.sh %U
```

File: com.blueokiris.bgrm.desktop [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+Icon=com.blueokiris.bgrm.png
```
Comment (hfiguiere) [inline com.blueokiris.bgrm.desktop:4]:
```suggestion
Icon=com.blueokiris.bgrm
```

File: com.blueokiris.bgrm.json [hunk 4]
```diff
@@ -0,0 +1,70 @@
+{
+    "app-id": "com.blueokiris.bgrm",
+    "runtime": "org.freedesktop.Platform",
+    "runtime-version": "21.08",
```
Comment (hfiguiere) [inline com.blueokiris.bgrm.json:4]:
```suggestion
    "runtime-version": "22.08",
```

File: com.blueokiris.bgrm.json [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    "sdk": "org.freedesktop.Sdk",
+    "command": "bgrm-fp.sh",
+    "finish-args": [
+        "--device=all",
+        "--socket=wayland",
+        "--socket=x11",
```
Comment (hfiguiere) [inline com.blueokiris.bgrm.json:10]: does it really support wayland? I yes, then it's `fallback-x11`

also missing share=ipc

File: com.blueokiris.bgrm.json [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+        "--filesystem=host:ro"
+    ], "modules": [
+        {
+            "name": "virtualenv",
+            "buildsystem": "simple",
+            "build-options": {
+                "build-args": [
+                    "--share=network"
+                ]
```
Comment (hfiguiere) [inline com.blueokiris.bgrm.json:19]: this is NOT allowed. you need to list the dependencies by hand.

File: com.blueokiris.bgrm.desktop (hunk 2 above)
Comment (blueOkiris) [reply com.blueokiris.bgrm.desktop]: Recommendation accepted

File: com.blueokiris.bgrm.desktop (hunk 3 above)
Comment (blueOkiris) [reply com.blueokiris.bgrm.desktop]: Corrected

File: com.blueokiris.bgrm.json (hunk 5 above)
Comment (blueOkiris) [reply com.blueokiris.bgrm.json]: It does not support wayland afaik.
I removed that and added the share ipc

File: com.blueokiris.bgrm.json (hunk 6 above)
Comment (blueOkiris) [reply com.blueokiris.bgrm.json]: Removed and replaced with individual python dependencies.

**Comment by blueOkiris** [comment]
bot, build com.blueokiris.bgrm

File: com.blueokiris.bgrm.json (hunk 4 above)
Comment (blueOkiris) [reply com.blueokiris.bgrm.json]: Upgraded to 22.08

**Review by hfiguiere — COMMENTED** [review]
I don't see an appstream file installed.

idealy the .desktop and icon should be submitted upstream.

File: com.blueokiris.bgrm.json (hunk 4 above)
Comment (blueOkiris) [reply com.blueokiris.bgrm.json]: In 22.08, setuptools is upgraded, which [causes issues with numpy](https://github.com/numpy/numpy/issues/22157)

I need numpy as a dependency for other things, which means as far as I can tell, my two options are:
- use the numpy .whl which is x86_64 exclusive
- use 21.08 with old setuptools so I can build numpy from source

**Comment by blueOkiris** [comment]
My application relies on mediapipe. There's not an aarch64 equivalent for that, so I can only support x86_64

**Comment by blueOkiris** [comment]
I may be able to build it from scratch, but that will take a while to flesh out. 
I'm also running into a million issues getting the dependencies working correctly.

Thus, I'm going to close this PR.

I'll open a new one once I've fixed everything. Thank you for helping me so far

**Pull request closed without merging**